test/themeutils.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) by Felix Weilbach <felix.weilbach@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+#include <QPaintDevice>
+#include <QTest>
+
+class FakePaintDevice : public QPaintDevice
+{
+public:
+    FakePaintDevice();
+
+    QPaintEngine *paintEngine() const override;
+
+    void setHidpi(bool value);
+
+protected:
+    int metric(QPaintDevice::PaintDeviceMetric metric) const override;
+
+private:
+    bool _hidpi = false;
+};
+
+#define TEST_THEME_MAIN(TestObject)              \
```

Comment:
Do we really need this macro? If that's because of the Q_INIT_RESOURCE calls, IIRC this could be done in the test class ctor.

Reply:
I didn't thought about that. Yes that would be better.